platform-core-module/src/main/java/com/softicar/platform/core/module/program/execution/cleanup/ProgramExecutionDeleter.java

```diff
@@ -0,0 +1,124 @@
+package com.softicar.platform.core.module.program.execution.cleanup;
+
+import com.softicar.platform.common.core.logging.Log;
+import com.softicar.platform.common.core.thread.sleep.Sleep;
+import com.softicar.platform.common.core.utils.DevNull;
+import com.softicar.platform.common.date.DateItemRange;
+import com.softicar.platform.common.date.Day;
+import com.softicar.platform.common.date.DayTime;
+import com.softicar.platform.core.module.program.AGProgram;
+import com.softicar.platform.core.module.program.execution.AGProgramExecution;
+import com.softicar.platform.core.module.program.execution.AGProgramExecutionLog;
+import com.softicar.platform.db.core.transaction.DbTransaction;
+import java.util.List;
+
+/**
+ * Deletes program execution records ({@link AGProgramExecution}) and their log
```

Comment:
- We could remove some redundancy here, like: `Deletes ({@link AGProgramExecution}) records and corresponding ({@link AGProgramExecutionLog}) records [...]`
- Also, there's `[...] records records [...]` in the Javadoc.

Reply:
done